lib/cookie.ts

```diff
@@ -383,8 +383,8 @@ function canonicalDomain(str: string | null) {
 
 // S5.1.3 Domain Matching
 function domainMatch(
-  str?: string,
-  domStr?: string,
+  str?: string | null,
+  domStr?: string | null,
```

Comment:
Based on the test cases, these methods seem to consider `null` to be valid input. Whether or not they _should_ going forward can be part of our upcoming review of how the library uses `null`/`undefined`.